tests/test_types.py

```diff
@@ -2774,3 +2776,81 @@ class Model(BaseModel):
         {'loc': ('my_none_dict', 'a'), 'msg': 'value is not None', 'type': 'type_error.not_none'},
         {'loc': ('my_json_none',), 'msg': 'value is not None', 'type': 'type_error.not_none'},
     ]
+
+
+@pytest.mark.parametrize(
+    'value,result',
+    (
+        ('1996-01-22', date(1996, 1, 22)),
+        (date(1996, 1, 22), date(1996, 1, 22)),
+    ),
+)
+def test_past_date_validation_success(value, result):
+    class Model(BaseModel):
+        foo: PastDate
+
+    assert Model(foo=value).foo == result
+
+
+@pytest.mark.parametrize(
+    'value,expected',
+    (
+        (date.today(), str(date.today())),
+        (date.today() + timedelta(1), str(date.today() + timedelta(1))),
+        (datetime.today(), str(date.today())),
+        (datetime.today() + timedelta(1), str(date.today() + timedelta(1))),
+    ),
+)
+def test_past_date_validation_fails(value, expected):
+    class Model(BaseModel):
+        foo: PastDate
+
+    with pytest.raises(ValidationError) as exc_info:
+        Model(foo=value)
+    assert exc_info.value.errors() == [
+        {
+            'loc': ('foo',),
+            'msg': f'date "{expected}" is not in the past',
+            'type': 'value_error.date.not_in_the_past',
+            'ctx': {'date': expected},
+        }
+    ]
+
+
+@pytest.mark.parametrize(
+    'value,result',
+    (
+        (date.today() + timedelta(1), date.today() + timedelta(1)),
+        (datetime.today() + timedelta(1), date.today() + timedelta(1)),
+    ),
+)
+def test_future_date_validation_success(value, result):
+    class Model(BaseModel):
+        foo: FutureDate
+
+    assert Model(foo=value).foo == result
+
+
+@pytest.mark.parametrize(
+    'value,expected',
+    (
+        (date.today(), str(date.today())),
+        (date.today() - timedelta(1), str(date.today() - timedelta(1))),
+        (datetime.today(), str(date.today())),
+        (datetime.today() - timedelta(1), str(date.today() - timedelta(1))),
```

Comment:
Please include one explicit case '2064-06-01' or something so we have a hardcoded case.

If pydantic is still alive in 2064, and anyone is still alive to use it, they can fix the test then. 😨